tools/ipfs_pinner.py

```diff
@@ -0,0 +1,80 @@
+#! /usr/bin/env python3
+
+import argparse
+import logging
+
+from logic.indexer_service import EventHandler
+from tools import db_utils
+from util import patches
+from util.contract import ContractHelper
+from util.ipfs import IPFSHelper
+assert patches
+
+
+def _ipfs_hashes_for_listings():
+    """
+    Returns an iterable with the IPFS hashes of all listings that exist on the
+    blockchain.
+    """
+    contract_helper = ContractHelper()
+    event_handler = EventHandler()
+    hashes = set()
+
+    def callback(payload):
+        address = event_handler._get_new_listing_address(payload)
+        data = event_handler._fetch_listing_data(address)
+        hashes.add(data['ipfs_hash'])
+
+    # TODO(cuongdo): extract 'NewListing(uint256)' into a variable, to avoid
+    # so much repetition.
+    contract_helper.fetch_events(["NewListing(uint256)"], callback)
```

Comment:
Fetching all events at once should be ok while we only have a few listings (hundreds or maybe thousands). But I suspect this will break when we have a larger volume of listing and we'll have to fetch chunks of events (maybe by specifying a block_from/block_to). Perhaps add a note about that for future ?